File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    missing = {}
    answer = []
    
    for i in range(len(name)):
        missing[name[i]] = yearning[i]
    
    for pic in photo:
        score = 0
        for people in pic:
            if people in missing.keys():
                score += missing[people]
        answer.append(score)
        
    return answer

File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    var scores = {};
    var answer = [];
    var score = 0;
    name.map((people, index) => scores[people] = yearning[index]);
    
    for(var ph of photo) {
        score = 0;
        for (var p of ph) {
            if (p in scores){ 
                score += scores[p];
            };
        }
        console.log(ph);
        console.log(score);
        answer.push(score);
    }
    return answer;
}